internal/onepassword/sdk/client.go

```diff
@@ -0,0 +1,180 @@
+package sdk
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	sdk "github.com/1password/onepassword-sdk-go"
+
+	"github.com/1Password/terraform-provider-onepassword/v2/internal/onepassword/model"
+)
+
+type Client struct {
+	sdkClient *sdk.Client
+}
+
+type SDKConfig struct {
+	ProviderUserAgent   string
+	ServiceAccountToken string
+	Account             string
+}
+
+func (c *Client) GetVault(ctx context.Context, uuid string) (*model.Vault, error) {
+	vault, err := c.sdkClient.Vaults().GetOverview(ctx, uuid)
+	if err != nil {
+		return nil, err
+	}
+
+	v := &model.Vault{}
+	v.FromSDKVault(&vault)
+
+	return v, nil
+}
+
+func (c *Client) GetVaultsByTitle(ctx context.Context, title string) ([]*model.Vault, error) {
+	vaultList, err := c.sdkClient.Vaults().List(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var result []*model.Vault
+	for _, vaultOverview := range vaultList {
+		fullVault, err := c.sdkClient.Vaults().GetOverview(ctx, vaultOverview.ID)
```

Comment:
Do we really need to get an overview?
It seems like both SDK methods returns the same `VaultOverview` struct
```
List(ctx context.Context, params ...VaultListParams) ([]VaultOverview, error)
GetOverview(ctx context.Context, vaultUuid string) (VaultOverview, error)
```

and it's possible to map to `model.Vault` from `vaultList, err := c.sdkClient.Vaults().List(ctx)`